rdt/transformers/categorical.py

```diff
@@ -460,8 +462,16 @@ def _transform(self, data):
                 Data to transform.
 
         Returns:
-            numpy.ndarray:
+            pd.Series.
         """
+        unseen_categories = set(pd.unique(data)) - set(self.categories_to_values.keys())
+        if unseen_categories:
+            warn(
+                f'Warning: The data contains new categories {unseen_categories} that were not '
+                'seen in the original data. Assigning them NaN values. If you want to model '
```

Comment:
I changed the message to say `Assigning them NaN values` instead of `Assigning them random values`, since that's the behavior of the code.

Reply:
I don't think it actually says in the issue but I believe we want to change the behavior to return a random value now instead. Is that right @npatki

Reply:
Ah yes, @fealho I implied the change in the warning message instead of explicitly stating it outright. We want to assign random values.

Rationale:
(a) not all ML methods can deal with null values; intended usage is not to introduce more nulls
(b) categorical transformers are treating nulls as a separate category; an unexpected category != null in that sense